### PR TITLE
Update voter tutorial

### LINKED ISSUE #432: Thoughts on the Voter Tutorial
Hi, just going through the voter tutorial, and I've noticed a couple things that bare mention.

 First it seems ```starport app github.com/joe-mcgee/voter``` scaffolded a slightly different project from the repo.  For example [this](https://github.com/cosmos/sdk-tutorials/blob/master/voter/voter/vue/src/views/Index.vue) popped out as 
```
<template>
  <div>
    <div class="sp-container">
      <sp-sign-in />
      <sp-bank-balances />
      <sp-token-send />
      <!-- this line is used by starport scaffolding # 4 -->
                <sp-type-form type="poll" :fields="['title', 'options', ]" module="voter" />
    </div>
  </div>
</template>

<script>
import * as sp from "@tendermint/vue";

export default {
  components: { ...sp },
};
</script>
```
amongst other differences, not sure where its pulling from.  Anyways, seems like an improvement over the old code for sure, just means that the tutorial instructions that work with Vue can be axed.

This means no specialized UI to add or remove options but https://github.com/cosmos/sdk-tutorials/blob/c16bc8edadc21a33b71b58e0bd9c792a4652ab68/voter/voter/x/voter/client/rest/txPoll.go#L13-L39

can be modified to
```Options []string``` -> ```Options string```
and split out into an array in the logic
```msg := types.NewMsgCreatePoll(creator, req.Title. strings.split(req.Options, " "))```

most certainly not good UI practice, but probably sufficient for such a tutorial

additionally it appears the ```starport type <x> <x_y> <x_z>``` doesn't appear to have just created just `types/MsgCreatePoll` but also `types/SetPoll` and `types/DeletePoll`  which adds some extra conversions of string -> []string.

Cheers,
Taylor
Dev @[Sifchain](https://sifchain.finance/)

### LINKED ISSUE #446: voter tutorial broken
first step `starport type poll title options` results in
```
> starport type poll title options
Error: poll type is already added.
Usage:
  starport type [typeName] [field1] [field2] ... [flags]

Flags:
  -h, --help                 help for type
      --module string        Module to add the type into. Default: app's main module
  -p, --path string          path of the app
      --sdk-version string   Target Cosmos-SDK Version -launchpad -stargate (default "launchpad")

poll type is already added.

```

Then in the other window a compilation error:
```
cannot build app:

	# github.com/keybase/go-keychain
cgo-gcc-prolog:203:11: warning: 'SecTrustedApplicationCreateFromPath' is deprecated: first deprecated in macOS 10.15 - No longer supported [-Wdeprecated-declarations]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Security.framework/Headers/SecTrustedApplication.h:59:10: note: 'SecTrustedApplicationCreateFromPath' has been explicitly marked deprecated here
# github.com/alice/voter/x/voter/client/cli
x/voter/client/cli/queryPoll.go:18:65: undefined: "github.com/alice/voter/x/voter/types".QueryListPoll
x/voter/client/cli/queryVote.go:18:65: undefined: "github.com/alice/voter/x/voter/types".QueryListVote
note: module requires Go 1.15
# github.com/alice/voter/x/voter/keeper
x/voter/keeper/poll.go:11:16: undefined: "github.com/alice/voter/x/voter/types".PollPrefix
x/voter/keeper/poll.go:19:55: undefined: "github.com/alice/voter/x/voter/types".PollPrefix
x/voter/keeper/vote.go:12:16: undefined: "github.com/alice/voter/x/voter/types".VotePrefix
x/voter/keeper/vote.go:20:54: undefined: "github.com/alice/voter/x/voter/types".VotePrefix
note: module requires Go 1.15
```

### PR DESCRIPTION
**Pull request by Tosch110**
Update voter tutorial to Starport v0.13.1

Closes #446 
Closes #432

### CODE REVIEW

**Review by fadeev — CHANGES_REQUESTED** [review]
The frontend has to be fixed so that it's usable both locally and on Gitpod.

## vue/src/components/PollForm.vue

```
<script>
import * as sp from "@tendermint/vue";
export default {
  components: { ...sp },
  data() {
    return {
      title: "",
      options: []
    };
  },
  methods: {
    add() {
      this.options = [...this.options, { title: "" }];
    },
    async submit() {
      const payload = {
        type: "poll",
        module: "voter",
        body: {
          title: this.title,
          options: this.options.map(o => o.title)
        }
      };
      await this.$store.dispatch("cosmos/entitySubmit", payload);
        await this.$store.dispatch("cosmos/entityFetch", payload);
    }
  }
};
</script>
```

Delete `vue/src/store/app.js`.

## vue/src/store/index.js

```
import Vue from "vue";
import Vuex from "vuex";
import axios from "axios";
import cosmos from "@tendermint/vue/src/store/cosmos.js";

Vue.use(Vuex);

export default new Vuex.Store({
  modules: { cosmos },
});
```

## vue/src/App.vue

```
<script>
export default {
  created() {
    this.$store.dispatch("cosmos/init");
    this.$store.dispatch("cosmos/entityFetch", {type: "poll", module: "voter"});
    this.$store.dispatch("cosmos/entityFetch", {type: "vote", module: "voter"});
  },
};
</script>
```

## vue/src/components/PollList.vue

```
<script>
import * as sp from "@tendermint/vue";
import AppRadioItem from "./AppRadioItem";
import AppText from "./AppText";
import {countBy } from "lodash"
export default {
  components: { AppText, AppRadioItem, ...sp },
  data() {
    return {
      selected: ""
    };
  },
  computed: {
    polls() {
      return this.$store.state.cosmos.data["voter/poll"] || [];
    },
    votes() {
      return this.$store.state.cosmos.data["voter/vote"] || [];
    }
  },
  methods: {
    results(id) {
      const results = this.votes.filter(v => v.pollID === id);
      return countBy(results, "value");
    },
    async submit(pollID, value) {
      const type = { type: "vote" };
      const body = { pollID, value };
      await this.$store.dispatch("cosmos/entitySubmit", { ...type, module:"voter", body });
      await this.$store.dispatch("cosmos/entityFetch", {...type, module: "voter"});
    }
  }
};
</script>
```

## vue/src/main.js

```
import Vue from "vue";
import App from "./App.vue";
import router from "./router";
import store from "./store";

Vue.config.productionTip = false;

new Vue({
  router,
  store,
  render: (h) => h(App),
}).$mount("#app");
```

Basically, we use `cosmos/entitySubmit` and pass `module: "voter"`. No longer need `app.js` and no changes to the store are required.

**Comment by fadeev** [comment]
@Tosch110 here is the finished state of the tutorial that works for me:

https://github.com/fadeev/starport/tree/develop/docs/voter

File: voter/voter/x/voter/client/cli/txPoll.go [hunk 1]
```diff
@@ -5,27 +5,71 @@ import (
...
 	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
-	"github.com/cosmos/sdk-tutorials/voter/voter/x/voter/types"
 )
 
 func GetCmdCreatePoll(cdc *codec.Codec) *cobra.Command {
 	return &cobra.Command{
 		Use:   "create-poll [title] [options]",
 		Short: "Creates a new poll",
-		Args:  cobra.MinimumNArgs(2),
+		Args:  cobra.ExactArgs(2),
```
[the remark this answers is not in the document]
Comment (Tosch110) [reply voter/voter/x/voter/client/cli/txPoll.go]: Addressed

File: voter/voter/x/voter/client/cli/txPoll.go [hunk 2]
```diff
@@ -5,27 +5,71 @@ import (
...
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+}
+
+func GetCmdSetPoll(cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "set-poll [id]  [title] [options]",
+		Short: "Set a new poll",
+		Args:  cobra.ExactArgs(3),
```
[the remark this answers is not in the document]
Comment (Tosch110) [reply voter/voter/x/voter/client/cli/txPoll.go]: Addressed

**Review by fadeev — APPROVED** [review]
Great work, @Tosch110! 🦖

**Pull request merged**